### PR TITLE
Fix handling of deployment properties

### LINKED ISSUE #14: Make sure deployer handles bucketed properties correctly
See https://github.com/spring-cloud/spring-cloud-dataflow/issues/754

`AppDeploymentRequest.deploymentProperties` should be used by the deployer and not passed on to the app

`AppDefinition.properties` should be passed to the app

### PR DESCRIPTION
**Pull request by trisberg**
- don't pass them to app
- count should be found in the deploymentProperties

Resolves #14

### CODE REVIEW

**Comment by sabbyanandan** [comment]
Tried this PR on a local build of SCDF-core and SCDF-Mesos with an updated `1.0.2.BUILD-SNAPSHOT` version of deployer artifacts. With the following stream, however, I still don't see 2 instances of `log-sink` application.

```
dataflow:>stream create --name words --definition "http | splitter --expression=payload.split(' ') | log"
Created new stream 'words'
dataflow:>stream deploy words --properties "app.splitter.producer.partitionKeyExpression=payload,app.log.count=2"
Deployed stream 'words'
```

**Comment by trisberg** [comment]
This relates to #17, so anything related to scaling/partitioning will be addressed in that PR. We can merge this as is or make all changes in the other PR.

File: src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java [hunk 1]
```diff
@@ -112,7 +111,7 @@ public String deploy(AppDeploymentRequest request) {
 
 		app.setCpus(cpus);
 		app.setMem(memory);
-		app.setInstances(Integer.getInteger(request.getDefinition().getProperties().get(COUNT_PROPERTY_KEY)));
+		app.setInstances(Integer.getInteger(request.getDeploymentProperties().get(COUNT_PROPERTY_KEY)));
```
Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java:114]: Ouch. Found the issue related to count not being passed properly and ... it hurts. `Integer.getInteger()` does not do what you'd think it does.

Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java:114]: Will fix, and I think an additional test is in order...

Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java:114]: Actually, test for instance count should go in abstrat deployer TCK. There are also opportunities for more tests now that deploymentProperties contract has been clarified. Will open another issue

**Comment by ericbottard** [comment]
Merged as 7dc2ae3, which includes a [fix](https://github.com/ericbottard/spring-cloud-mesos-deployer/commit/bb09e1dce94eb9b2338e3266b0670a11c3b67c60) for count handling

**Pull request closed without merging**